pallets/attesters/src/lib.rs

```diff
@@ -313,118 +467,139 @@ pub mod pallet {
-                        AttestationStatus::Pending
-                    } else {
-                        AttestationStatus::Approved
-                    };
+                // Update the status of the batch
+                let quorum = (T::ActiveSetSize::get() * 2 / 3) as usize;
+                let full_approval = T::ActiveSetSize::get() as usize;
+                if batch.signatures.len() >= quorum {
+                    log::debug!(
+                        "Batch {:?} is ready for submission by majority",
+                        batch.message_hash()
+                    );
+                }
+                if batch.signatures.len() >= full_approval {
+                    batch.status = BatchStatus::ReadyForSubmissionFullyApproved;
+                    log::debug!(
+                        "Batch {:?} is ready for submission by full approval",
+                        batch.message_hash()
+                    );
+                    Self::deposit_event(Event::NewConfirmationBatch(target, batch.clone()));
+                }
 
-                    sfx_attestation.for_ = attestation_for;
-                    sfx_attestation.status = status;
+                Self::deposit_event(Event::AttestationSubmitted(account_id));
 
-                    Attestations::<T>::insert(sfx_hash, sfx_attestation.clone());
+                Ok::<(), DispatchError>(())
+            })?;
 
-                    let mut target_batches = Batches::<T>::get(target).unwrap_or_default();
-                    let current_block = frame_system::Pallet::<T>::block_number();
+            Ok(())
+        }
 
-                    if target_batches.is_empty() {
-                        // Create a new batch if there are no batches yet
-                        let new_batch =
-                            Batch::new(target, frame_system::Pallet::<T>::block_number());
-                        target_batches.push(new_batch);
-                    } else {
-                        let is_full = target_batches.last().map_or(false, |b| {
-                            b.attestations.len() >= T::MaxBatchSize::get() as usize
-                        });
-                        let interval_passed = target_batches.last().map_or(false, |b| {
-                            current_block - b.created >= T::BatchingWindow::get()
-                        });
-
-                        if is_full || interval_passed {
-                            // Create a new batch
-                            let new_batch =
-                                Batch::new(target, frame_system::Pallet::<T>::block_number());
-                            target_batches.push(new_batch);
-                        }
-                    }
+        #[pallet::weight(10_000)]
+        pub fn commit_batch(
+            origin: OriginFor<T>,
+            target: TargetId,
+            target_inclusion_proof_encoded: Vec<u8>, // Add this parameter to accept Ethereum batch hash
```

Comment:
I think since this is specific to ethereum, it should allow some enum to switch off of the target since we would also:
- know the size of the arguments, reducing attack vector on unbounded vectors running around the system

Reply:
opened issue for adjusting per different architectures https://github.com/t3rn/t3rn/issues/996